python-sdk/src/langwatch/workflow/__init__.py

```
@@ -0,0 +1,184 @@
+    path_segments = [seg for seg in urlparse(run_url).path.split("/") if seg]
+    if "experiments" in path_segments:
+        idx = path_segments.index("experiments")
+        if idx + 1 < len(path_segments):
+            return path_segments[idx + 1]
+    return ""
+
+
+def run(
+    workflow_id: str,
+    *,
+    data: Optional[Union[List[dict], "pd.DataFrame"]] = None,
```

Ruff (0.15.17)

[error] Undefined name `pd`

(F821)